modular_skyrat/modules/microfusion/code/microfusion_energy_master.dm

```diff
@@ -0,0 +1,807 @@
+#define DUALWIELD_PENALTY_EXTRA_MULTIPLIER 1.4
+
+// Master file for cell loadable energy guns. PROCS ONLY YOU MONKEYS!
+// This file is a copy/paste of _energy.dm with extensive modification.
+
+/obj/item/gun/microfusion
+	name = "prototype detatchable cell energy projection aparatus"
+	desc = "The coders have obviously failed to realise this is broken."
+	icon = 'modular_skyrat/modules/microfusion/icons/microfusion_gun40x32.dmi'
+	icon_state = "mcr01"
+	bayonet_icon = 'modular_skyrat/modules/microfusion/icons/microfusion_gun40x32.dmi'
+	gunlight_icon = 'modular_skyrat/modules/microfusion/icons/microfusion_gun40x32.dmi'
+	lefthand_file = 'modular_skyrat/modules/microfusion/icons/guns_lefthand.dmi'
+	righthand_file = 'modular_skyrat/modules/microfusion/icons/guns_lefthand.dmi'
+	has_gun_safety = TRUE
+	can_flashlight = FALSE
+	can_bayonet = FALSE
+	w_class = WEIGHT_CLASS_BULKY
+
+	/// What type of power cell this uses
+	var/obj/item/stock_parts/cell/microfusion/cell
+	var/cell_type = /obj/item/stock_parts/cell/microfusion
+	///if the weapon has custom icons for individual ammo types it can switch between. ie disabler beams, taser, laser/lethals, ect.
```

Comment:
```suggestion
	/// If the weapon has custom icons for individual ammo types it can switch between. ie disabler beams, taser, laser/lethals, etc.
```